raster_feeder/steps/rotate.py

```diff
@@ -60,11 +62,54 @@ def extract_region(path):
 
         # read precipitation
         variable = nc.variables['precipitation']
-        prcp = variable[:]
-        fillvalue = variable._FillValue.item()
 
-    # copy out a region of interest for member selection
-    y_slice, x_slice = config.STATISTICS_ROI
+        # NetCDF performs linear scaling and masking automatically
+        nc.set_auto_maskandscale(True)
+        prcp = variable[:]
+        fillvalue = np.finfo(prcp.dtype).max
+
+        # NetCDF will produce a MaskedArray if there are masked pixels
+        if isinstance(prcp, np.ma.MaskedArray):
+            prcp = prcp.filled(fillvalue)
+
+    # transform the region of interest to indices in the array
+
+    # create the geometry
+    x1, y1, x2, y2 = config.ROI_ESPG32756
+    ring = ogr.Geometry(ogr.wkbLinearRing)
+    poly = ogr.Geometry(ogr.wkbPolygon)
+    ring.AddPoint(x1, y1)
+    ring.AddPoint(x2, y1)
+    ring.AddPoint(x2, y2)
+    ring.AddPoint(x1, y2)
+    ring.AddPoint(x1, y1)
+    poly.AddGeometry(ring)
+    poly.AssignSpatialReference(EPSG32756)
+
+    # transform the geometry to native projection
+    target = osr.SpatialReference()
+    target.ImportFromProj4(str(config.PROJECTION))
+    poly.TransformTo(target)
+    x1_proj, x2_proj, y1_proj, y2_proj = poly.GetEnvelope()
+
+    # transform the envelope to indices
+    inv_geo_transform = gdal.InvGeoTransform(config.GEO_TRANSFORM)
+    x1_px, y1_px = gdal.ApplyGeoTransform(inv_geo_transform, x1_proj, y1_proj)
+    x2_px, y2_px = gdal.ApplyGeoTransform(inv_geo_transform, x2_proj, y2_proj)
+
+    # swap the indices if necessary
```

Comment:
Maar is het nodig? Dit is een heel specifiek geval en elke keer hetzelfde. Dus als het moet dan altijd en anders mag het van mij weg.